tests/testthat/test-shinytest2-tm_data_table.R

```diff
@@ -44,6 +44,7 @@ test_that("e2e - tm_data_table: Verify checkbox displayed over data table", {
 })
 
 test_that("e2e - tm_data_table: Verify module displays data table", {
+  testthat::skip_if_not_installed("rvest")
```

Comment:
I have recently pushed changes into the `teal` due to noSuggest check and the class definition looks like this:
https://github.com/insightsengineering/teal/blob/7eac4d58a5373aa511f363a2f259d850749501f7/R/TealAppDriver.R#L10-L20

Just thinking loud now, would it help if we would have the following instead?
```
    if (!requireNamespace("shinytest2", quietly = TRUE)) {
      if (requireNamespace("testthat", quietly = TRUE) && getNamespace("testthat")$is_testing()) {  
       getNamespace("testthat")$skip("rvest is not installed")
      } else {
        stop("Please install 'shinytest2' package to use this class.")
      }
    }
```

My motivation is to avoid a lot of skip statements in the upstream packages like here.

Reply:
That's awesome, I'll create the PR in {teal} to do just this. I think the `$is_testing()` is not required as `skip()` throws an error and it's on testthat to capture it and handle